modules/hitlnext/src/backend/service.ts

```diff
@@ -0,0 +1,100 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+
+import { Config } from '../config'
+import { MODULE_NAME } from '../constants'
+import { ExitTypes, IHandoff } from '../types'
+import { StateType } from '.'
+import Repository from './repository'
+
+export const toBpDestination = (botId: string, handoff: Pick<IHandoff, 'userId' | 'userThreadId' | 'userChannel'>) => {
+  return { botId, target: handoff.userId, threadId: handoff.userThreadId, channel: handoff.userChannel }
+}
+
+class Service {
+  constructor(private bp: typeof sdk, private state: StateType, private repository: Repository, private realtime) {}
+
+  async createHandoff(
+    botId: string,
+    dest: Pick<IHandoff, 'userId' | 'userThreadId' | 'userChannel' | 'status'>,
+    timeoutDelay: number
+  ) {
+    const config: Config = await this.bp.config.getModuleConfigForBot(MODULE_NAME, botId)
+    const eventDestination = toBpDestination(botId, dest)
+    const attributes = await this.bp.users.getAttributes(dest.userChannel, dest.userId)
+    const language = attributes.language
+
+    const handoff = await this.repository.createHandoff(botId, dest).then(handoff => {
+      this.state.cacheHandoff(botId, handoff.userThreadId, handoff)
+      return handoff
+    })
+
+    if (config.transferMessage) {
+      await this.sendMessageToUser(config.transferMessage, eventDestination, language)
+    }
+
+    this.realtime.sendPayload(botId, { resource: 'handoff', type: 'create', id: handoff.id, payload: handoff })
+
+    if (timeoutDelay !== undefined && timeoutDelay > 0) {
+      setTimeout(async () => {
+        const userHandoff = await this.repository.getHandoff(handoff.id)
+
+        if (userHandoff.status === 'pending') {
+          await this.updateHandoff(userHandoff, botId, { status: 'expired' })
+          await this.transferToBot(eventDestination, 'timedOutWaitingAgent')
+        }
+      }, timeoutDelay * 1000)
+    }
+
+    return handoff
+  }
+
+  async resolveHandoff(handoff: IHandoff, botId: string, payload) {
+    const eventDestination = toBpDestination(botId, handoff)
+    const updated = await this.updateHandoff(handoff, botId, payload)
+    await this.transferToBot(eventDestination, 'handoffResolved')
+
+    return updated
+  }
+
+  async updateHandoff(handoff: Partial<IHandoff>, botId: string, payload: any) {
```

Comment:
How does this update the handoff? Can you add comments, I find it more complex than simply an update on the handoff itself.

Reply:
It updates the handoff with the provided payload. I didn't create the module and I don't have the time to refactor it, I just moved some pieces of the code from the api file to a new service file, to break it into smaller pieces of code which can be reused more efficiently.

The next person which will have to code something in the module can continue this cleanup

Reply:
No need to refactor, you can simply add a comment